docs/smart-contracts/views.md

```diff
@@ -2,86 +2,194 @@
-            </ul>
-          </li>
-        </ul>
-      </td>
-    </tr>
-  </tbody>
-</table>}
-
-Another contract might provide an `equalizeWith` entrypoint such that if they have more tokens than another user, they can make their balances equal (plus or minus one if the total amount is odd).
-The following example code for this contract takes advantage of the `getBalance(user)` view of the first contract: to determine the balance of each user:
+Views are a way for contracts to expose information to other contracts and to off-chain consumers.
+
+Views help you get around a limitation in smart contracts: a smart contract can't access another contract's storage.
+Smart contracts can provide information via callbacks, but using a callback means calling entrypoints, which is an asynchronous action.
+
+By contrast, views are synchronous; a contract can call a view and use the information that it returns immediately.
+
+Like entrypoints, views can accept parameters, access the contract's storage, and call other views.
+Unlike entrypoints, views return a value directly to the caller.
+However, views can't cause side effects, so they can't create operations, including calling smart contracts and transferring tez.
+Views also can't change the contract storage.
+
+Nodes can run views without creating any operations, which lets off-chain users get a node to run a view and return the result immediately, as a convenient way to get information from a smart contract.
+
+## Examples
+
+Views can provide information about tokens.
+You can use views to provide an account's balance of a token type or the total amount of a token in circulation.
+
+DEXs can provide the exchange rate between two tokens or the amount of liquidity in the pool.
+
+Instead of repeating certain logic in multiple places, you can put the logic in a view and use it from different smart contracts.
+
+## Creating views in JsLIGO
+
+Views in LIGO look like entrypoints because they receive the input values and storage as parameters, but they have the `@view` annotation instead of the `@entry` annotation.
+They return a value instead of a list of operations and the new value of the storage.
+
+This JsLIGO view returns the larger of two numbers:
+
+```ts
+type get_larger_input = [int, int];
+
+@view
+const get_larger = (input: get_larger_input, _s: storage): int => {
+  const [a, b] = input;
+  if (a > b) {
+    return a;
+  }
+  return b;
+}
+```
+
+This view returns a value from a big-map in storage:
+
+```ts
+type storageType = big_map<string, string>;
+
+@view
+const get = (key: string, s: storageType): string => {
+  const valOpt = Big_map.find_opt(key, s);
+  return match(valOpt) {
+    when(Some(val)): val;
+    when(None): "";
+  }
+}
+```
+
+## Calling views in JsLIGO
+
+This JsLIGO code calls the `get_larger` view from the previous example by passing the target contract address, parameters, and view name to the `Tezos.call_vew()` function:
```

Comment:
The call example could be much more meaningful on the get view (instead of the dummy get_larger view), as it really gets some useful info from the contract.

Reply:
I'll work on a better example.